roles/openshift_aws/tasks/vpc_and_subnet_id.yml

```diff
@@ -15,8 +15,14 @@
   ec2_vpc_subnet_facts:
     region: "{{ openshift_aws_region }}"
     filters:
-      "availability-zone": "{{ openshift_aws_subnet_az }}"
       vpc-id: "{{ vpcout.vpcs[0].id }}"
+      "tag:Name": "{{ item }}"
+  with_items: ['{%- if (openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list | length) == 1 -%}
+      {{ openshift_aws_subnet_az }}
+    {%- elif (openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list | length) > 1 -%}
+      {{ openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list }}
+    {%- else -%}
+    {%- endif -%}']
   register: subnetout
```

Comment:
<img width="800" alt="screen shot 2018-08-01 at 1 32 41 pm" src="https://user-images.githubusercontent.com/2479463/43538257-80b340d6-958f-11e8-9012-57a510d889ab.png">

Reply:
@mwoodson @sdodson @joelddiaz

Reply:
Simply using ...
```{{ openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list }}```
here would be awesome.

That would abandon the default_az subkey in openshift_aws_vpc, key openshift_aws_subnet_az, and filter get_default_az.  I would love to get rid of all those bits as long as we all agree.

@abutcher in regards to cluster-operator, what do you think about removing default_az and using element 0 as default az?

Reply:
I like the explicit default az attribute but I don't feel very strongly about it.

Reply:
If we remove default az will CO be affected at all?

Reply:
what do you mean by "remove default az"?

Reply:
He means remove the `default_az` field we set here: https://github.com/openshift/cluster-operator/blob/c2f5837cd1ac8b8ed752d42eb5d3dca42ababec4/pkg/ansible/generate.go#L44

I believe our default is driven by an arg here https://github.com/openshift/cluster-operator/blob/489a78eb1bf85d43964d3de61654aee4e71e6f08/cmd/aws-machine-controller/machine-controller.go#L52 so I don't think removing `default_az` will impact us.

Reply:
We use the `default_az` in the inventory when provisioning the ELB since we rely on openshift-ansible for that. So that's the only place it may impact us.

Reply:
re deploying ELBs ... it would just use element 0 for single az deployments